src/ser.rs

```diff
@@ -55,6 +55,21 @@ where
     Ok(output)
 }
 
+/// Same as `to_bytes` but use `limit` as max container depth instead of MAX_CONTAINER_DEPTH
```

Comment:
It may be worth better documenting that `MAX_CONTAINER_DEPTH` is still respected and that a passed in limit lowers the maximum depth to a point below `MAX_CONTAINER_DEPTH` and that you cannot move it higher than `MAX_CONTAINER_DEPTH`

Reply:
I wonder if we want this by the way. It feels a bit artificial. @zekun000 @bmwill, any thought?

Reply:
i feel it's still good to have some hard coded high limit to avoid arbitrary large numbers